### PR TITLE
#2582 Adding a datadog helm template for datadog api key

### PR DESCRIPTION
**Pull request by nelsestu**
This template is now available for any other containers that need to use the datadog rest api

## What was the problem?

We need the ep-merge app container to receive a secret as an environment variable. Other containers will want to use the same secret in the future, so a global shared template makes the most sense.

Associated tickets or Slack threads:
- #2582

## How does this fix it?[^1]
Adds the global shared template _datadog.tpl and then propagates that into the ep-merge deployment.

## How to test this PR
We just need to validate the values all come out correctly post helm deploy to dev.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: helm/_shared/named_templates/_datadog.tpl [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  valueFrom:
+    secretKeyRef:
+      name: vro-datadog
+      key: DATADOG_API_KEY
+- name: DATADOG_API_KEY_ID
+  valueFrom:
+    secretKeyRef:
+      name: vro-datadog
+      key: DATADOG_API_KEY_ID
+{{- end }}
```
Comment (msnwatson) [inline helm/_shared/named_templates/_datadog.tpl]: Nit: new line

**Review by msnwatson — APPROVED** [review]
LGTM with nit

File: helm/_shared/named_templates/_datadog.tpl (hunk 1 above)
Comment (nelsestu) [reply helm/_shared/named_templates/_datadog.tpl]: no you are totally right to call that.  a fix for that is now in flight running checks.

**Review by chengjie8 — APPROVED** [review]
LG

**Review by tejans24 — APPROVED** [review]

**Pull request merged**